app/parsers/bulkrax/csv_parser_decorator.rb

```diff
@@ -0,0 +1,146 @@
+    end
+    # rubocop:enable Metrics/AbcSize, Metrics/MethodLength
+
+    private
+
+    def coerce_unpacked_files!
+      combine_csvs
+      combine_files
+      coerce_csv
+      clean_up_upacked_files
+    end
+
+    # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+    def combine_csvs
+      combined_csv_path = File.join(importer_unzip_path, COMBINED_CSV_FILENAME)
+      csv_file_paths = Dir.glob(File.join(importer_unzip_path, '*.csv'))
+
+      # Collect all unique headers from all CSV files, starting with file and model since they do not originally exist
+      all_headers = csv_file_paths.each_with_object(%w[file model]) do |file_path, headers|
+        CSV.foreach(file_path, headers: true) do |row|
+          headers << row.headers
+        end
+      end.uniq.flatten
+
+      # Open the combined CSV file for writing
+      CSV.open(combined_csv_path, 'wb') do |combined_csv|
+        combined_csv << all_headers
+
+        csv_file_paths.each do |file_path|
+          CSV.foreach(file_path, headers: true) do |row|
+            combined_row = all_headers.index_with { |_header| nil }
+            row.to_h.each { |key, value| combined_row[key] = value if combined_row.key?(key) }
+            combined_csv << combined_row.values
+          end
+        end
+      end
+    end
+    # rubocop:enable Metrics/AbcSize, Metrics/MethodLength
+
+    # rubocop:disable Metrics/MethodLength
+    def combine_files
+      file_paths = Dir.glob(File.join(importer_unzip_path, '*'))
+      file_paths.select! { |file_path| Dir.exist?(file_path) }
+      files = []
+      file_paths.each do |file_path|
+        files += Dir.glob(File.join(file_path, '*'))
+      end
+
+      combined_files_dir = File.join(importer_unzip_path, COMBINED_FILES_DIRNAME)
+      Dir.mkdir(combined_files_dir) unless File.directory?(combined_files_dir)
+
+      files.each do |file|
+        FileUtils.mv(file, combined_files_dir)
+      end
+    end
+    # rubocop:enable Metrics/MethodLength
+
+    # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/MethodLength, Metrics/PerceivedComplexity
+    def coerce_csv
+      csv_path = File.join(importer_unzip_path, COMBINED_CSV_FILENAME)
+      files = Dir.glob(File.join(importer_unzip_path, COMBINED_FILES_DIRNAME, '*'))
+                 .map { |path| File.basename(path) }
+                 .reject { |file| file.downcase.include?('thumbnail') }
+
+      csv_rows = CSV.foreach(csv_path, headers: true, return_headers: true).map(&:to_hash)
+
+      # Separate the header from the rest of the rows
+      header = csv_rows.shift
+      new_rows = csv_rows.each do |row|
+        file_entry = files.select { |file| file.include?(row['ss_pid'].tr(':', '_')) }.join(',')
+        row['file'] = file_entry if file_entry.present?
+        if (row['bs_isCommunity'] || row['bs_isCollection'])&.downcase == 'true' || in_collection_and_community?(row)
+          row['model'] = Hyrax.config.collection_model
```

Comment:
If this does go back to Bulkrax, we should not invoke `Hyrax` directly here

Reply:
We decided that after their initial migration, we would be removing this code and expect them to use Bulkrax standards. So the only part relevant to Bulkrax would be if we wanted to allow tar files in addition to zip.